images/devtools-terraform-v1beta1/context/Makefile

```diff
@@ -104,6 +104,29 @@ validate-fix: | .terraform
 # utility
 ########################################################################
 
+GH_CLI := $(shell command -v gh 2> /dev/null)
+github_oauth_token=
+terraform_switcher_version=
+terraform_switcher_tag:="latest"
+ifdef terraform_switcher_version
+	terraform_switcher_tag:="tags/$(terraform_switcher_version)"
+endif
+ifdef github_oauth_token
+	USE_GITHUB_OAUTH_TOKEN:="-U \"$(github_oauth_token):x-oauth-basic\""
+endif
+ifndef GH_CLI
+	terraform_switcher_release:=curl --header "Accept: application/vnd.github.v3+json" $(USE_GITHUB_OAUTH_TOKEN) "https://api.github.com/repos/warrensbox/terraform-switcher/releases/$(terraform_switcher_tag)"
+else
+	terraform_switcher_release:=gh api --header "Accept: application/vnd.github.v3+json" "/repos/warrensbox/terraform-switcher/releases/$(terraform_switcher_tag)"
```

Comment:
I'm unsure why we can't just use curl, what benefit does depending on github CLI provide?

Reply:
Automation, that's it. If you and others feel this is useless - then I will abandon the PR and we are done :)

Reply:
How is GitHub CLI more automated than curl? You use curl just above it. I'm not saying it is useless, I'm just not sure what value it adds. The curl command actually works fine without a github token aslo (it will just get rate limited), so why have both? Why not just make the github token optional and if it is not set don't pass it in the header? This just makes it complicated and more work to maintain as far as I can tell.

Reply:
If you just want to go with github cli then I'm fine with that also, but still if it is one of the following commands:

```
gh api --header "Accept: application/vnd.github.v3+json" "/repos/warrensbox/terraform-switcher/releases/latest"
```
vs
```
curl --header "Accept: application/vnd.github.v3+json" "https://api.github.com/repos/warrensbox/terraform-switcher/releases/latest"
```
Then from the actual commands appearance they are both the same to me, and one does not seem more automated than the other.

Reply:
Now that I think about it though I guess the benefit is that if you are logged in with gh cli then it will inject the token, and if that is the rationale, then I would say it is worth it, but that does not make it more automated.